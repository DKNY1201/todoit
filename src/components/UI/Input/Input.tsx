import * as React from 'react';

interface IInput {
    placeholder: string;
    value?: string;
    changed?: () => void;
}

const Input = ({placeholder, value, changed}: IInput) => (
    <div className="input-container">
        <input placeholder={placeholder} value={value} onChange={changed} />
    </div>
);

export default Input;