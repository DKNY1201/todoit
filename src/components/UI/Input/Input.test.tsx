import * as enzyme from 'enzyme';
import * as React from 'react';

import Input from './Input';

describe('<Input />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = enzyme.shallow(<Input placeholder="Enter Todo" value="A Todo"/>);
    });

    it('should render a icon correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render an input with correct placeholder', () => {
        expect(wrapper.find('input').prop('placeholder')).toEqual('Enter Todo');
    });

    it('should have an input with correct value', () => {
        expect(wrapper.find('input').prop('value')).toEqual('A Todo');
    });

    it('should call provided function when value changes', () => {
        const changed = jest.fn();
        wrapper.setProps({changed});
        expect(changed).toHaveBeenCalledTimes(0);
        wrapper.find('input').simulate('change');
        expect(changed).toHaveBeenCalledTimes(1);
    });
});