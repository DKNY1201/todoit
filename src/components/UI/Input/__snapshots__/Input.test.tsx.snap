// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render a icon correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    placeholder="Enter Todo"
    value="A Todo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={undefined}
        placeholder="Enter Todo"
        value="A Todo"
/>,
      "className": "input-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": undefined,
        "placeholder": "Enter Todo",
        "value": "A Todo",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={undefined}
          placeholder="Enter Todo"
          value="A Todo"
/>,
        "className": "input-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "placeholder": "Enter Todo",
          "value": "A Todo",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
